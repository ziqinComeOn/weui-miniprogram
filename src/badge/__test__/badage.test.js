import path from 'path'
import simulate from 'miniprogram-simulate'

describe('badge', () => {
    let id

    beforeAll(() => {
        id = simulate.load(path.resolve(__dirname, '../badge'), { less: true })
    })

    test('badge with content', () => {
        const content = 'badge content'
        const comp = simulate.render(id, {
            content: content
        })
        comp.attach(document.createElement('parent-wrapper'))

        expect(comp.toJSON()).toMatchSnapshot()
    })

    test('badge without content', () => {
        const comp = simulate.render(id, {})
        comp.attach(document.createElement('parent-wrapper'))

        expect(comp.toJSON()).toMatchSnapshot()
    })
})
