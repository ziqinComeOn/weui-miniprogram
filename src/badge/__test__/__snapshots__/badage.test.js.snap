// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`badge badge with content 1`] = `
<main>
  <wx-view
    class="weui-badge  "
  >
    badge content
  </wx-view>
</main>
`;

exports[`badge badge without content 1`] = `
<main>
  <wx-view
    class="weui-badge  weui-badge_dot"
  >
    
  </wx-view>
</main>
`;
